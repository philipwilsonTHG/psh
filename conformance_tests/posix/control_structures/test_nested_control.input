# Test nested control structures
# PSH supports arbitrarily nested control structures (v0.19.0 feature)

# Test nested if statements
score=85
if [ "$score" -ge 90 ]; then
    echo "Grade A"
elif [ "$score" -ge 80 ]; then
    echo "Grade B"
    if [ "$score" -ge 85 ]; then
        echo "High B"
    else
        echo "Low B"
    fi
else
    echo "Grade C or lower"
fi

# Test nested loops
echo "Nested for loops:"
for i in 1 2; do
    echo "Outer loop: $i"
    for j in a b; do
        echo "  Inner loop: $j"
        if [ "$i" = "2" ] && [ "$j" = "b" ]; then
            echo "    Special case: $i$j"
        fi
    done
done

# Test while loop with nested if
echo "While with nested if:"
count=1
while [ "$count" -le 3 ]; do
    echo "Count: $count"
    if [ "$count" -eq 2 ]; then
        echo "  Middle value"
        if [ "$count" -lt 3 ]; then
            echo "    Not the last"
        fi
    fi
    count=$((count + 1))
done

# Test case with nested constructs
echo "Case with nested constructs:"
value="test"
case "$value" in
    test)
        echo "Found test"
        for item in 1 2; do
            echo "  Processing: $item"
        done
        ;;
    other)
        echo "Found other"
        ;;
    *)
        echo "Unknown"
        ;;
esac