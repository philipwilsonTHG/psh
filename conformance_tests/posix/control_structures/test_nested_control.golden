Grade B
High B
Nested for loops:
Outer loop: 1
  Inner loop: a
  Inner loop: b
Outer loop: 2
  Inner loop: a
  Inner loop: b
    Special case: 2b
While with nested if:
Count: 1
Count: 2
  Middle value
    Not the last
Count: 3
Case with nested constructs:
Found test
  Processing: 1
  Processing: 2
